### ISSUE TITLE
LUA Stack when loading  US Flatlands Map 4x 2.6.0.0 

### ISSUE THREAD

**Issue by matze0301** [GMNGjoy/FS25_ContractBoost#25]
Hi,

i have only the map and ContractBoost in the mod drawer and did not load macDon Pack.

LUA stack while first time loading, it never reaches 100% and stalls at "loading vehicles"

```
2024-12-16 16:20 -- ContractBoost :: loadMapData
2024-12-16 16:20 -- ContractBoost:MissionBalance :: settings updated.
2024-12-16 16:20 -- ContractBoost:MissionBalance :: scaleMissionReward
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: rewardFactor:1.5
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 1: baleMission | updated 2200 => 3500
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 2: baleWrapMission | updated 300 => 500
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 3: plowMission | updated 2800 => 4200
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 4: cultivateMission | updated 2300 => 3450
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 5: sowMission | updated 2000 => 3000
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 6: harvestMission | updated 2500 => 3750
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 7: hoeMission | updated 1500 => 2250
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 8: weedMission | updated 2000 => 3000
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 9: herbicideMission | updated 1500 => 2250
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 10: fertilizeMission | updated 1500 => 2250
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 11: mowMission | updated 2500 => 3750
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 12: tedderMission | updated 1500 => 2250
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 13: stonePickMission | updated 2200 => 3300
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 14: deadwoodMission | skipped, not found on map
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 15: treeTransportMission | skipped, not found on map
2024-12-16 16:20 ---- ContractBoost:MissionBalance :: Mission 16: destructibleRockMission | skipped, not found on map
2024-12-16 16:20 -- ContractBoost:MissionBalance :: scaleMissionReward complete.
2024-12-16 16:20 -- ContractBoost:MissionBorrow :: fillMissionTools
2024-12-16 16:20 Error: Running LUA method 'update'.
G:/LS25/Modordner/US-Flatlands-4x/FS25_ContractBoost/scripts/MissionBorrow.lua:45: invalid argument #1 to 'pairs' (table expected, got nil)

```
Please let me know if you need the full log :-)
Map DL: https://forbidden-mods.de/filebase/file/57477-us-flatlands-map-4x/

**Comment by GMNGjoy** [comment]
@matze0301 thank you for taking the time to submit a report!

One missing piece from the full log that is critical for me to help you - what version of _Contract Boost_ are you running? (this sounds similar to an issue that was already reported with version `1.0.0.0`, and fixed in `1.0.0.2` which was finally released on ModHub.

**Comment by GMNGjoy** [comment]
## _not a bug_

the problem here is that map maker in their hasty conversion has disabled ALL MISSION VEHICLES, which is _not in any way_ a problem of the _Contract Boost_ mod.

you can easily fix it if you want, by unzipping that map, opening the `map/map.xml` file, and finding this block:

```xml
<!--<missions vehicleFilename="$dataS/missionVehicles.xml">
        <deadwoodMission treeType="DEADWOOD">
            <spots filename="map/missions/deadwoodSpots.i3d" />
        </deadwoodMission>
        <treeTransportMission treeType="TRANSPORT">
            <spots filename="map/missions/transportSpots.i3d" />
        </treeTransportMission>
        <destructibleRockMission>
            <spots filename="map/missions/destructibleRocksSpots.i3d" />
            <marker filename="$data/shared/assets/marker/rockMissionFlag.i3d" />
        </destructibleRockMission>
    </missions>-->
```
and updating it to look like this:
```xml
<missions vehicleFilename="$dataS/missionVehicles.xml" />
<!--        <deadwoodMission treeType="DEADWOOD">
            <spots filename="map/missions/deadwoodSpots.i3d" />
        </deadwoodMission>
        <treeTransportMission treeType="TRANSPORT">
            <spots filename="map/missions/transportSpots.i3d" />
        </treeTransportMission>
        <destructibleRockMission>
            <spots filename="map/missions/destructibleRocksSpots.i3d" />
            <marker filename="$data/shared/assets/marker/rockMissionFlag.i3d" />
        </destructibleRockMission>
    </missions>-->
```
then zipping it back up, which will fix the root issue (the lack of contract vehicles). 

That said, I've also pushed / released [1.0.3.1](https://github.com/GMNGjoy/FS25_ContractBoost/releases/tag/1.0.3.1) which should prevent against incorrect map configurations causing an issue with the _Contract Boost_ mod.

**Comment by matze0301** [comment]
Wow, thank you for finding the problem and providing a solution :-)
